File: clojure/bob/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn shouting? [message]
  (and
    (= (str/upper-case message) message)
    (not= (str/lower-case message) message)))



(defn response-for [message] ;; <- arglist goes here
  (let [trimmedMessage (str/trim message)]
    (cond
      (= trimmedMessage "") "Fine. Be that way!"
      (shouting? trimmedMessage) "Whoa, chill out!"
      (str/ends-with? trimmedMessage "?") "Sure."
      :else "Whatever.")))

File: go/bob/bob.go
package bob

import "strings"

const testVersion = 3

// Hey emulates Bob's responses
func Hey(message string) string {
	trimmedMessage := strings.TrimSpace(message)
	if trimmedMessage == "" {
		return "Fine. Be that way!"
	}
	if strings.ToUpper(trimmedMessage) == trimmedMessage && strings.ToLower(trimmedMessage) != trimmedMessage {
		return "Whoa, chill out!"
	}
	if strings.HasSuffix(trimmedMessage, "?") {
		return "Sure."
	}
	return "Whatever."
}
